### PR TITLE
Add a couple of examples writing 3D data

### LINKED ISSUE #237: Documentation: 6.3 Coordinates and Geometry Types
https://github.com/Toblerity/Fiona/blob/master/docs/manual.rst#63coordinates-and-geometry-types

Is it worth adding a couple of examples to this section, and perhaps putting something about '3D' in the section title? It took me a while to figure out what I was doing wrong yesterday (trying to write 3D geometries to a `'Point'` layer, instead of  a `'3D Point'` layer).

### PR DESCRIPTION
**Pull request by chadwhawkins**
Resolves #237

### CODE REVIEW

File: docs/manual.rst [hunk 1]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+---------------------------------
 
 If you write 3D coordinates, ones having (x, y, z) tuples, to a 2D file
 ('Point' schema geometry, for example) the z values will be lost.
 
+.. sourcecode:: python
+
+  schema_props = OrderedDict([('foo', 'str')])
+
+  record = {
```
Comment (sgillies) [inline docs/manual.rst]:
```suggestion
  feature = {
```

File: docs/manual.rst [hunk 2]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+      'properties': OrderedDict([
+          ('foo', 'bar')
+      ])
+  }
+
+  with fiona.open(
+          '/tmp/file.shp',
+          'w',
+          driver='ESRI Shapefile',
+          schema={'geometry': 'Point', 'properties': schema_props}) as c:
```
Comment (sgillies) [inline docs/manual.rst]:
```suggestion
          schema={'geometry': 'Point', 'properties': schema_props}) as collection:
```

File: docs/manual.rst [hunk 3]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+      c.write(record)
+
+  with fiona.open('/tmp/file.shp') as c:
```
Comment (sgillies) [inline docs/manual.rst]:
```suggestion
  with fiona.open('/tmp/file.shp') as collection:
```

File: docs/manual.rst [hunk 4]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+          ('foo', 'bar')
+      ])
+  }
+
+  with fiona.open(
+          '/tmp/file.shp',
+          'w',
+          driver='ESRI Shapefile',
+          schema={'geometry': 'Point', 'properties': schema_props}) as c:
+      c.write(record)
```
Comment (sgillies) [inline docs/manual.rst]:
```suggestion
      c.write(feature)
```

File: docs/manual.rst [hunk 5]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+
+  with fiona.open('/tmp/file.shp') as c:
+      print(c[0]['geometry'])
```
Comment (sgillies) [inline docs/manual.rst]:
```suggestion
      print(next(collection)['geometry'])
```

File: docs/manual.rst [hunk 6]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+      'properties': OrderedDict([
+          ('foo', 'bar')
+      ])
+  }
+
+  with fiona.open(
+          '/tmp/file.shp',
+          'w',
+          driver='ESRI Shapefile',
+          schema={'geometry': '3D Point', 'properties': schema_props}) as c:
```
Comment (sgillies) [inline docs/manual.rst]:
```suggestion
          schema={'geometry': '3D Point', 'properties': schema_props}) as collection:
```

File: docs/manual.rst [hunk 7]
```diff
@@ -1044,15 +1044,64 @@ or an ordered dict.
...
+      c.write(record)
+
+  with fiona.open('/tmp/file.shp') as c:
+      print(c[0]['geometry'])
```
Comment (sgillies) [inline docs/manual.rst]: etc.

**Comment by chadwhawkins** [comment]
Thanks for the review @sgillies! Made the suggested changes and will go back and run through with black soon.

**Comment by snorfalorpagus** [comment]
@chadwhawkins Looks good to me. Just needs running through the formatter.

**Comment by chadwhawkins** [comment]
Whoops, sorry this fell off my radar. Thanks for bringing it back to my attention @snorfalorpagus. Ran the `python` blocks through the black formatter but `pycon` blocks would need some scripting. Hopefully it's ok to leave those for now @sgillies?

**Comment by sgillies** [comment]
@chadwhawkins no need to do anything about the pycon blocks. Thanks so much!

**Pull request merged**